picard/util/alphabet_detector.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# The MIT License (MIT)
+#
+# Copyright (C) 2015 Eli Finkelshteyn
+# Copyright (C) 2021 Vladislav Karbovskii
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import unicodedata as ud
+
+
+def detect_alphabet(unistr):
+    return set(ud.name(char).split(' ')[0]
+               for char in unistr if char.isalpha())
```

Comment:
Just a thought: Maybe an improvement would be to not only return a set, but a weighted list depending on the number of occurrences for a script. E.g. Japanese names sometimes also contain Latin characters. If you had configured artist_locale_exception for latin that name would still be considered latin. But if we would weight the scripts by occurence we could say e.g. primary script is "HIRAGANA", even though it also contains "LATIN".

Reply:
This could also provide some intelligent basis for selecting which alias to display if we eventually allow multiple script/char sets to be enabled.